android/app/src/main/java/app/organicmaps/MwmApplication.java

```diff
@@ -275,22 +270,11 @@ public boolean arePlatformAndCoreInitialized()
     System.loadLibrary("organicmaps");
   }
 
-  // Called from JNI.
-  @Keep
-  @SuppressWarnings("unused")
-  private void forwardToMainThread(final long taskPointer)
-  {
-    Message m = Message.obtain(mMainLoopHandler, () -> nativeProcessTask(taskPointer));
-    m.obj = mMainQueueToken;
-    mMainLoopHandler.sendMessage(m);
-  }
```

Comment:
Is it the core reason to refactor? runLater is better?
We have a phantom bug, that also can be caused by worker -> UI thread forwarding:
https://github.com/organicmaps/organicmaps/issues/9397

Reply:
1. Move thread-related code into the correct class.
2. These messages and loopers are unnecessary, especially in application class.
3. Make initPlatform static - class object is not needed in GUI thread class because all methods are static right now
4. Pass context object in initPlatform to reduce class cohesion

`runLater` is same thing but could potentially work faster.

This refactoring most likely won't fix the bug. I've already replied in the issue why this bug happens (from my pov)